libs/@blockprotocol/graph/src/types/temporal-versioning.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Types used in embedding applications and blocks that support multi-axis temporal versioning schemes.
+ */
+
+import { Subtype } from "../util.js";
+
+/**
+ * An ISO 8601 formatted timestamp string
+ */
+export type Timestamp = string;
+
+/**
+ * @todo - doc
+ */
+export type TemporalAxes = "transactionTime" | "decisionTime";
+
+/**
+ * The bound of a time-interval that is either exclusively or inclusively limited by a `Timestamp`
+ */
+export type TimestampLimitedTemporalBound = {
+  kind: "inclusive" | "exclusive";
+  limit: Timestamp;
+};
```

Comment:
Just for comparison: In Rust we use `"included"` and `"excluded"` as `bound` and `timestamp` instead of `limit`. I'm not requesting to change this, just want to flag it as this is subject to discussion. Personally, I like `kind` and `limit`, but prefer `"included"` and `"excluded"`. To deviate from those we could also use `"open"` or `"closed"`.

Reply:
I was going to open a comment to highlight this as well but had to jump off yesterday, thank you!

I guess the open question we'd like answered as part of merging this PR, is do we prefer a `kind` of:

1.`"included" | "excluded"` or,
2. `"inclusive" | "exclusive"`

I prefer "inclusive" and "exclusive", it aligns more with what I think I've heard more commonly in the past. "Inclusive bounds"

Reply:
cc @CiaranMn @thehabbos007

Reply:
"inclusive" / "exclusive" is definitely more familiar to me

Reply:
Also worth highlighting another change I've made from our other implementation. I describe the endpoints of the intervals here as the `start` and `end` bounds of the time intervals as I feel that this is more intuitive when talking about ranges of time.

We have opted for something more general in the other code, with `lower` and `upper` bounds.

Not sure if there are any other thoughts on that.

Reply:
I also prefer `"inclusive" | "exclusive"` as it would describe the `kind` of `Bound`. 

`"open" | "closed"` also works, but I don't feel strongly about it.

Reply:
>  I describe the endpoints of the intervals here as the start and end bounds of the time intervals as I feel that this is more intuitive when talking about ranges of time.
> 
> We have opted for something more general in the other code, with lower and upper bounds.

Personally, I find `start` and `end` more general than `lower` and `upper`. `lower` and `upper` encapsulate an ordering, `lower` is strictly less or equal to `upper`, while the `start` could greater than `end` - depending on your point of view. Maybe this is fine for time-related things as it's unusual, that a time interval starts in the present and ends in the past, but I think that `lower` and `upper` is more precise here.

Reply:
Slightly prefer `start` and `end`, not a very strong feeling as anyone using them already has to make their way through the concept of multi-axis versioning and can probably deal with `upper` and `lower`

Reply:
Judging from the feelings in this thread I'm going to opt to leave it as is
- `start` and `end`
- `"inclusive"` and `"exclusive"`